src/Raven.Client/Documents/Subscriptions/SubscriptionState.cs

```diff
@@ -27,7 +29,9 @@ public class SubscriptionState : IDatabaseTask, IDatabaseTaskStatus
         // the responsible node of the subscription,
         // in sharding context - orchestrator node tag
         public string NodeTag { get; set; }
-        
+
+        public ArchivedDataProcessingBehavior ArchivedDataProcessingBehavior { get; set; }
```

Comment:
This isn't nullable? What if you have a subscription definition from 5.4 version?

Reply:
https://github.com/poissoncorp/ravendb/blob/9f9727fb5eb2aad640068c7b53a5505141e848f2/src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs#L60

any other usages are covered by this line https://github.com/poissoncorp/ravendb/blob/9f9727fb5eb2aad640068c7b53a5505141e848f2/src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs#L60

`SubscriptionState.ArchivedDataProcessingBehavior` comes from `PutSubscriptionCommand`

Reply:
ensured backward compatibility